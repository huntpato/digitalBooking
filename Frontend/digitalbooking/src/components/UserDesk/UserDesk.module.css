.userDesk{
    display: flex;
    flex-direction: column;
    min-height: 93vh;
    padding: 68px 0px 0px 0px;
    width: 100%;
}

.menu{
    align-self: center;
    background-color: var(--background-white);
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin: 40px 0px 4px 0px;
    width: 90vw;
}

.menu button{
    background-color: var(--background-white);
    border: none;
    border-radius: 5px;
    cursor: pointer;
    padding: 10px 3px 10px 0px;
    margin: 3px 10px 4px 0px;
    text-align: center;
    width:28%;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
}

.desk{
    align-self: center;
    background-color:var(--background-white);
    display: flex;
    flex-direction: column;
    align-content: center;
    justify-content: center;
    margin: 30px 0px 30px 0px;
    width: 90vw;
}

.menu i{
    color: var(--principal-color);
    padding:0 8px 0 0px ;
    margin:0;
}

@media (max-width: 480px){

    .menu{
        flex-direction: column;
        align-items: center;
        margin: 20px 0px 4px 0px;
    }

    .menu button{
        padding: 10px 0px 10px 10px;
        margin: 3px 0px 4px 0px;
        text-align: left;
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
    }
}
