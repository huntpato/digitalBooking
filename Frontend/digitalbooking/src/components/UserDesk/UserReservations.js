import React, {useState, useEffect} from 'react'
import { ReServices } from '../ReServices/ReServices';
import { Link } from "react-router-dom";
import { CancelReservation } from './CancelReservation/CancelReservation'
import { NotFound } from '../Products/NotFound/NotFound';
import diffDates from "../../functions/diffDates"
import styles from './UserReservations.module.css'
import userReserv from './userReserv';

export const UserReservations = () => {

    const {userReservations, cardContainer, container, buttonsBooking} = styles;
    const [cards, setCards] = useState([]);
    const [loading, setLoading]= useState(true);
    const user= JSON.parse(localStorage.getItem("data")).name;
    
    /*let lStyles = {
      width: '50%',
      padding: '0px 0px 0px 0px',
      margin: '0px 0px 0px 0px',
      backgroundColor: 'red'
    };*/
    
    userReserv.then(products => {
      setCards(products);
      setLoading(false);
    })


    
    return (
        <div className= { userReservations }> 
         {loading && <h5>Cargando reservas</h5>}   
         {!loading && cards.length == 0 ? <NotFound/>: <h5>{`Reservas de ${user}`}</h5>}
         {cards && cards.map((booking)=>(
           <div className={container}>
             <div className={cardContainer}>
              <h5>{`Fecha: desde el ${booking.reservationStartDate} hasta el ${booking.reservationEndDate}`}</h5>
              <hr/> 
              <p>{booking.product.productCategory.description}</p>
              <h4>{booking.product.productName}</h4>
              <img src={booking.product.photoGallery[0]} alt="Fotografia del producto"/>
              <ReServices locServices={booking.product.services} />
              <p>{booking.product.productDescription}</p>
              <hr/>
              <span>{`$${booking.product.pricePerNight} / Noche`}</span>
              <span>{`$${booking.product.pricePerNight * (diffDates(new Date(booking.reservationStartDate), new Date(booking.reservationEndDate)))} / Total por ${diffDates(new Date(booking.reservationStartDate), new Date(booking.reservationEndDate))} Noches`}</span>
              {/*<p></p>*/}
              <div className={ buttonsBooking }>
                <Link to={`/product/${booking.product.id}`}>{" "}<button>Ver m√°s</button></Link>
                <CancelReservation id={booking.id} />
              </div>
             </div>
           </div> 
         ))}
        </div>
    )
}
