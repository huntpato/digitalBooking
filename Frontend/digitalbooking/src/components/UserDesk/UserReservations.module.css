.userReservations{
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    gap: 2%;
    padding: 0px 0 0 0px;
    margin: 0px 0 0 0px;
    width: 90vw;
    align-items: center;
}

.userReservations h5{
    margin: 0px 0px 20px 0px;
    font-size: 18px;
    width: 100%;
}

.container{
    align-self: flex-start;
    background-color: var(--background-white);
    border-radius: 6px;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 0px 0 30px 0px;
    box-shadow: 1px 1px 4px 2px rgba(0, 0, 0, 0.05);
    width:32%;
    
}

.cardContainer{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-evenly;
    margin: 15px 0px 10px 0px;
    width:90%;
    min-height: 590px;   
}

.cardContainer h5{
    margin: 0px 0px 10px 0px;
    font-size: 14px;
    width: 100%;
}

.cardContainer hr{
    width: 100%;
    margin: 0px 0px 0px 0px;
}

.cardContainer p{
    font-size: 12px;
    margin: 14px 0px 10px 0px;
    width: 100%;
}

.cardContainer h4{
    margin: 0px 0px 10px 0px;
    overflow:hidden;
    text-overflow:ellipsis;    
    white-space:nowrap;
    width: 100%;
}

.cardContainer img{
    width:100%;
    height: 260px;
    border-radius: 8px;
    margin: 5px 0px 20px 0px;
}

.buttonsBooking{
    margin: 15px 0px 0px 0px;
    width:100%;
}

.buttonsBooking button{
    background-color: var(--principal-color);
    border-radius: 5px;
    border: none;
    color: var(--background-white);
    cursor: pointer;
    height: 30px;
    margin: 0px 0px 10px 0px;
    width: 100%;
}

.cardContainer span{
    font-size: 12px;
    font-weight: 600;
    margin: 10px 0px 0px 0px;
    width: 100%;
}

@media (max-width: 768px){

    .userReservations{
        align-self: center;
        flex-direction: column;
        justify-content: space-between;
        width: 80vw;
        gap: 0%;
    }

    .container{
        width:100%;   
    }
    
    .cardContainer{
        margin: 25px 0px 20px 0px;
        min-height: 0px;
    }
    
    .cardContainer img{
        width:100%;
        height: 290px;
        border-radius: 8px;
        margin: 5px 0px 20px 0px;
    }

}

@media (max-width: 480px){

    .userReservations{
        width: 90vw;
    }

    .cardContainer img{
        height: 265px;
    }


}