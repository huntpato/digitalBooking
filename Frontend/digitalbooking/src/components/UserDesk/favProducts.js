const url2 = `https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/products`;
const user = JSON.parse(localStorage.getItem("data"));
const bearer = user && `${JSON.parse(localStorage.getItem("bearer")).jwt}`;
  const options = {
    headers: {
      Authorization: bearer,
      "Content-Type": "application/json",
      "X-Requested-With": "XMLHttpRequest",
    },
  };


async function favProducts() {
    let response = [];
    let data2 = [];  
    try{ 
       response = await (await fetch(url2, options)).text()
       data2= JSON.parse(await response);
    }
    catch(error) {
        console.log(error);
    };

    return data2;
}

export default favProducts();