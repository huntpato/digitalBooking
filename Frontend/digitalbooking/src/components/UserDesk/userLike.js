const { user } = useContext(UserContext);
const userId = user && JSON.parse(user).id;
const url2 = `https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/stats/users/${userId}`;;
const bearer = `${JSON.parse(localStorage.getItem("bearer")).jwt}`;
  const options = {
    headers: {
      Authorization: bearer,
      "Content-Type": "application/json",
      "X-Requested-With": "XMLHttpRequest",
    },
  };


async function favProducts() {
    let response = [];
    let data2 = [];  
    try{ 
       response = await (await fetch(url2, options)).text()
       data2= JSON.parse(await response);
    }
    catch(error) {
        console.log(error);
    };

    return data2;
}

export default favProducts();