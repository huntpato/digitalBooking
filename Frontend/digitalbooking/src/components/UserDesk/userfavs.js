const user = JSON.parse(localStorage.getItem("data"))
const userId= user && user.id;
const url2 =  `https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/stats/users/${userId}`;
const bearer = user && `${JSON.parse(localStorage.getItem("bearer")).jwt}`;
  const options = {
    headers: {
      Authorization: bearer,
      "Content-Type": "application/json",
      "X-Requested-With": "XMLHttpRequest",
    },
  };


async function userfavs() {
    let response = [];
    let data2 = [];  
    try{ 
       response = await (await fetch(url2, options)).text()
       data2= JSON.parse(await response);
    }
    catch(error) {
        console.log(error);
    };

    return data2;
}

export default userfavs();