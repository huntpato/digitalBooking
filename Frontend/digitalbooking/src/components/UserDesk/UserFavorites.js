import React, { useContext, useEffect, useState } from "react";
import { UserContext } from "../UserContext/UserContext";
import { ReCard } from "../ReCard/ReCard";
import styles from "./userFavorites.module.css";
import useFetch from "use-http";
import favProducts from "./favProducts";
import userfavs from "./userfavs";
import { NotFound } from "../Products/NotFound/NotFound";

export const UserFavorites = () => {
  const [data, setData]= useState();
  const [dataFiltered, setDataFiltered] = useState();
  const [filtrable, setFiltrable] = useState();
  const [liked, setliked]= useState([]);
  const [prodCard, setprodCard] = useState([]);
  const { userFavorites } = styles;
  const [data2, setdata2] = useState([]);
 

  useEffect(() => {
    favProducts.then(products => {
      setdata2(products);
    })
  
    userfavs.then(products=>{
      setData(products)
    })  
  }, [data2])
  
  useEffect(() => {
    let filtered;
    if(data != undefined){
    filtered = data.filter(function(element){
      return element.like == true;
    });
    setDataFiltered(filtered)
  }
  }, [data2, data])

  useEffect(() => {
    let userfavs=[];
    if(dataFiltered != undefined && data != undefined){
      dataFiltered.forEach(element => {
       userfavs.push(element.productId)
    });
    setliked(userfavs);
  };
  }, [dataFiltered, data])

  useEffect(() => {
    let filteredProd= []
    if(liked != [] && data2 != undefined){
      filteredProd= data2.filter(item => liked.includes(item.id))
      setprodCard(filteredProd)
      setFiltrable(true)
  }}, [liked])
 
  return (
    <div className={userFavorites}>
      {!filtrable && <h3>Cargando Favoritos...</h3>}
      {filtrable && prodCard[0]? 
        prodCard.map((product, index) => (
          <ReCard
            key={index}
            type={product.productCategory.name}
            title={product.productName}
            img={product.photoGallery[0]}
            alt={product.productDescription}
            text={product.productDescription}
            loc={product.location}
            rate={product.review}
            id={product.id}
            services={product.services}
            latlong={product.coordinates}
          />
        )): <h5>Todavia no tenes favoritos.</h5>}
    </div>
  );
};
