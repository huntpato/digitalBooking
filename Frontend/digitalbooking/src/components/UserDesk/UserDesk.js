import React, {  useContext, useState, useMemo } from 'react'
import { useHistory } from 'react-router-dom'
import { UserContext } from "../UserContext/UserContext";
import { UserReservations } from './UserReservations';
import { ProductBanner } from '../ProductBanner/ProductBanner';
import styles from './UserDesk.module.css'
import { Forbidden } from '../Forbidden/Forbidden';
import { UserFavorites } from './UserFavorites';
 

export const UserDesk = () => {
    const {user, setuser } = useContext(UserContext);
    const userName=  user && JSON.parse(localStorage.getItem("data")).name;
    const [udesk, setUdesk]= useState();
    const value = useMemo(
        () => ({ udesk, setUdesk }),
        [udesk, setUdesk]
    );
    const { userDesk, menu, desk } = styles; 
    const [reservation, setreservation]= useState(false);
    const [fav, setFav] = useState(false);
    let history = useHistory();  
   
    const handleClick = (e) => {
        e.preventDefault();
        setuser(null);
        localStorage.clear();
        history.push("/");
    };

    const handleReservation = (e)=> {
        e.preventDefault();
        setreservation(true);
        setFav(false);
        setUdesk("reservation")
    }

    const handleFavorites = (e)=> {
        e.preventDefault();
        setFav(true);
        setreservation(false);
        setUdesk("favorites")
    }

    let linkStyles = {
        backgroundColor: 'white',
        borderRadius: '5px',
        boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.15)',
        color: 'black',
        textDecoration: 'none',
        textAlign: 'center',
        width: '28%',
        padding: '10px 0px 10px 0px',
        margin: '3px 0px 4px 0px',
      };



    return (
        <div className= { userDesk }>
            <ProductBanner title="Perfil de" subtitle={userName} /> 
            <div className={ menu }>
                <button onClick={ handleReservation }> <i className="fas fa-calendar-check"></i> Mis reservas</button>
                <button onClick= { handleFavorites }><i className="fas fa-heart"></i>Favoritos</button>
                <button onClick={handleClick} style={linkStyles}><h5> <i className="fas fa-sign-out-alt"></i> Cerrar sesi√≥n</h5></button>
            </div>
            <div className={ desk }>
                {reservation && <UserReservations />}
                {fav && <UserFavorites />}
            </div>
        </div>
    )
}
