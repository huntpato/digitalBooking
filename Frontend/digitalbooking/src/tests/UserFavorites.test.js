import React from "react";
import { shallow, mount } from "enzyme";
import  {UserFavorites} from "../components/UserDesk/UserFavorites";
import "@testing-library/jest-dom";
import { expect } from "@jest/globals";

describe("Pruebas en componente <UserFavorites />", () => {
  const wrapper = mount(<UserFavorites />);

  test("Comprobar que <UserFavorites /> renderiza correctamente.", () => {
    expect(wrapper).toMatchSnapshot();
  });

 
});
