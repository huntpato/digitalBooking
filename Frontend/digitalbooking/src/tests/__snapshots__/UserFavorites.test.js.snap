// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <UserFavorites /> Comprobar que <UserFavorites /> renderiza correctamente. 1`] = `
<div
  className="userFavorites"
>
  <h5>
    Todavia no tenes favoritos.
  </h5>
</div>
`;
